plugin/pkg/admission/security/podsecuritypolicy/admission_test.go

```diff
@@ -2119,30 +2118,109 @@ func TestGetMatchingPolicies(t *testing.T) {
-		pspInformer := informerFactory.Extensions().InternalVersion().PodSecurityPolicies()
-		store := pspInformer.Informer().GetStore()
-		for _, psp := range v.inPolicies {
-			store.Add(psp)
-		}
+		var (
+			oldPod     *kapi.Pod
+			shouldPass = v.expectedPolicy != ""
+			authz      = &TestAuthorizer{usernameToNamespaceToAllowedPSPs: v.allowed}
+			canMutate  = true
+		)
+		pod := goodPod()
+		pod.Namespace = v.ns
+		pod.Spec.ServiceAccountName = v.sa
+		testPSPAdmitAdvanced(k, kadmission.Create, v.inPolicies, authz, v.user,
+			pod, oldPod, shouldPass, shouldPass, canMutate, v.expectedPolicy, t)
+	}
+}
 
-		authz := &TestAuthorizer{usernameToNamespaceToAllowedPSPs: v.allowed}
-		allowedPolicies, err := getMatchingPolicies(pspInformer.Lister(), v.user, v.sa, authz, v.ns)
-		if err != nil {
-			t.Errorf("%s got unexpected error %#v", k, err)
-			continue
-		}
-		allowedPolicyNames := sets.NewString()
-		for _, p := range allowedPolicies {
-			allowedPolicyNames.Insert(p.Name)
-		}
-		if !v.expectedPolicies.Equal(allowedPolicyNames) {
-			t.Errorf("%s received unexpected policies.  Expected %#v but got %#v", k, v.expectedPolicies.List(), allowedPolicyNames.List())
-		}
+func TestPolicyAuthorizationErrors(t *testing.T) {
+	policyWithName := func(name string) *extensions.PodSecurityPolicy {
+		p := restrictivePSP()
+		p.Name = name
+		return p
+	}
+
+	const (
+		sa       = "sa"
+		ns       = "test"
+		userName = "user"
+	)
+
+	tests := map[string]struct {
+		priviliged           bool
+		inPolicies           []*extensions.PodSecurityPolicy
+		allowed              map[string]map[string]map[string]bool
+		expectValidationErrs int
+	}{
+		"policies not allowed": {
+			allowed: map[string]map[string]map[string]bool{},
+			inPolicies: []*extensions.PodSecurityPolicy{
+				policyWithName("policy1"),
+				policyWithName("policy2"),
+			},
+			expectValidationErrs: 0,
+		},
+		"policy allowed by user": {
+			allowed: map[string]map[string]map[string]bool{
+				"user": {
+					"test": {"policy1": true},
+				},
+			},
+			inPolicies: []*extensions.PodSecurityPolicy{
+				policyWithName("policy1"),
+				policyWithName("policy2"),
+			},
+			expectValidationErrs: 1,
+		},
+		"policy allowed by service account": {
+			allowed: map[string]map[string]map[string]bool{
+				serviceaccount.MakeUsername("test", "sa"): {
+					"test": {"policy2": true},
+				},
+			},
+			inPolicies: []*extensions.PodSecurityPolicy{
+				policyWithName("policy1"),
+				policyWithName("policy2"),
+			},
+			expectValidationErrs: 1,
+		},
+		"multiple policies allowed": {
+			allowed: map[string]map[string]map[string]bool{
+				"user": {
+					"test": {"policy1": true},
+				},
+				serviceaccount.MakeUsername("test", "sa"): {
+					"test": {"policy2": true},
+				},
+			},
+			inPolicies: []*extensions.PodSecurityPolicy{
+				policyWithName("policy1"),
+				policyWithName("policy2"),
+			},
+			expectValidationErrs: 2,
+		},
+	}
+	for desc, tc := range tests {
+		t.Logf(desc)
```

Comment:
Is it intentionally or just forgotten? If the first, then how it could be useful?

Reply:
'Intentional. I find it much neater than adding something like `"TestCase = %s", desc` to every assertion. But actually I think `t.Run` is the ideomatic way of doing this, so I switched to that.